.github/workflows/ci.yml

```
@@ -195,6 +195,36 @@ jobs:
       if: ${{ matrix.task.run-if }}
 
 
+  pre-commit:
+    name: Check pre-commit integration
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
```

```suggestion
    - uses: actions/checkout@v4
```
This should be done across the board really since v3, but that's can be saved for a separate PR :)